### PR TITLE
Wallet: send token from drawer

### PR DESCRIPTION
**Pull request by OmarBasem**
This PR implements sending a token from the token item drawer.

Demo:

https://github.com/status-im/status-mobile/assets/29354102/f871e1d8-71d8-47d2-9d4f-277113d154a4

### CODE REVIEW

**Comment by J-Son89** [comment]
@OmarBasem, @briansztamfater - is the mechanics under the hood working as expected for this? 🤔 
e.g in the previous route where click 'send' button then selects the receiver and then select asset where as here it is the other way around. Have we thought about how we want to handle navigation etc for the two different paths? 🤔

**Comment by OmarBasem** [comment]
@J-Son89  The flow is the same, except that when going to the send flow from a token's drawer the select asset screen will be skipped as it has already been selected.

Maybe to make them consistent, in the default send flow, selecting the token could be done before selecting the destination address. I think other apps do it that way, and it would be consistent with when entering the send flow from a token's drawer.

**Comment by J-Son89** [comment]
_> @J-Son89 The flow is the same, except that when going to the send flow from a token's drawer the select asset screen will be skipped as it has already been selected._
ok great, yes this is what I was wondering. Sounds good 👍 


_> Maybe to make them consistent, in the default send flow, selecting the token could be done before selecting the destination address. I think other apps do it that way, and it would be consistent with when entering the send flow from a token's drawer._
I don't think we need to change this, this would be something to ask the designers, I was mostly curious if the code was setup correctly to handle this already (I can't see that you need to make adjustments here to handle different flows etc)
and sounds like it all works as expected - 
i.e it's worth doing a test, if you take this path is the send feature still working?

**Comment by OmarBasem** [comment]
@J-Son89 yes tested them, both paths work the same

File: src/status_im/contexts/wallet/send/events.cljs [hunk 1]
```diff
@@ -45,15 +45,24 @@
...
    {:db (assoc-in db [:wallet :ui :send :token] token)
     :fx [[:navigate-to-within-stack [:wallet-send-input-amount stack-id]]]}))
 
+(rf/reg-event-fx :wallet/send-select-token-drawer
+ (fn [{:keys [db]} [{:keys [token]}]]
+   {:db (assoc-in db [:wallet :ui :send :token] token)}))
+
+(rf/reg-event-fx :wallet/clean-selected-token
+ (fn [{:keys [db]}]
+   {:db (assoc-in db [:wallet :ui :send :token] nil)}))
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/send/events.cljs]: Can we dissoc `:token` instead of setting a `nil` value?

**Review by J-Son89 — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]
Nice work! 🚀

**Review by mmilad75 — APPROVED** [review]
Nice

**Review by vkjr — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hey @OmarBasem could you resolve the conflicts, please?

**Comment by OmarBasem** [comment]
> Hey @OmarBasem could you resolve the conflicts, please?

Done!

**Comment by qoqobolo** [comment]
Thank you @OmarBasem, PR can be merged!

**Pull request merged**